player.c

```diff
@@ -512,6 +523,7 @@ static void _producer_buffer_fill_update(void)
 /* 		d_print("\n"); */
 		player_info_priv.buffer_fill = fill;
 		player_info_priv.buffer_fill_changed = 1;
+		player_info_changed();
```

Comment:
Yeah, looks like a hack to be honest. And I think this can happen both too fast and too slow. Why did you decide to remove the timeout?